packages/admin-panel/src/VizBuilderApp/components/PreviewSection.js

```diff
@@ -141,11 +163,12 @@ export const PreviewSection = () => {
       setFetchEnabled(false);
     },
     vizType,
+    previewMode: getTab(tab).previewMode,
   });
-  const [tab, setTab] = useState(0);
 
   const handleChange = (event, newValue) => {
     setTab(newValue);
+    setFetchEnabled(true);
```

Comment:
Now we have to re-fetch when switching tabs, as the chart preview and the data preview will always fetch slightly different data.